### PR TITLE
CMake: Place tests where everything else is

### PR DESCRIPTION
**Pull request by glebm**
This should hopefully fix DLL lookup on Windows (#3778)

@tsunamistate @obligaron Can you please see if this resolves the issue?

### CODE REVIEW

**Comment by obligaron** [comment]
Cleaned build-folder, rebuild, run tests => Everything works fine. Thanks. 🙂

File: Brewfile [hunk 1]
```diff
@@ -2,4 +2,5 @@ brew "cmake"
 brew "fmt"
 brew "sdl2"
 brew "libsodium"
+brew "googletest"
```
Comment (AJenbo) [inline Brewfile:5]: already appied

Comment (glebm) [reply Brewfile]: Rebased

**Review by AJenbo — APPROVED** [review]

**Comment by tsunamistate** [comment]
It does solve the issue of tests not executing on Debug, but it still attempts and fails to link `libdevilutionx_so.dll` in Release build. For some reason, it tried to link it in `/test` folder

```
[447/471] Linking CXX shared library libdevilutionx_so.dll
  FAILED: libdevilutionx_so.dll test/libdevilutionx_so.lib 
  cmd.exe /C "cmd.exe /C ""C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E __create_def D:\src\devilutionX\build\x64-Release\test\CMakeFiles\libdevilutionx_so.dir\.\exports.def D:\src\devilutionX\build\x64-Release\test\CMakeFiles\libdevilutionx_so.dir\.\exports.def.objs && cd D:\src\devilutionX\build\x64-Release" && "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_dll --intdir=test\CMakeFiles\libdevilutionx_so.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100203~1.0\x86\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100203~1.0\x86\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx86\x64\link.exe /nologo @CMakeFiles\libdevilutionx_so.rsp  /out:libdevilutionx_so.dll /implib:test\libdevilutionx_so.lib /pdb:libdevilutionx_so.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO  /INCREMENTAL:NO /LTCG /DEF:test\CMakeFiles\libdevilutionx_so.dir\.\exports.def  && cmd.exe /C "cd /D D:\src\devilutionX\build\x64-Release\test && "C:\Program Files\PowerShell\7\pwsh.exe" -noprofile -executionpolicy Bypass -file D:/src/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary D:/src/devilutionX/build/x64-Release/libdevilutionx_so.dll -installedDir D:/src/devilutionX/build/x64-Release/vcpkg_installed/x64-windows/bin -OutVariable out""
  [448/471] Linking CXX executable devilutionx.exe
  ninja: build stopped: subcommand failed.

Build All failed.
```

**Comment by glebm** [comment]
@tsunamistate Can you try to get a more detailed error message? The above just says "FAILED" and "subcommand failed"

**Pull request merged**